src/Contracts/Request.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RicorocksDigitalAgency\Soap\Contracts;
+
+use RicorocksDigitalAgency\Soap\Response\Response;
+use RicorocksDigitalAgency\Soap\Support\Header;
+
+interface Request
+{
+    public function to(string $endpoint): self;
+
+    /**
+     * @param array<mixed> $parameters
+     *
+     * @return mixed
+     */
+    public function __call(string $name, array $parameters);
+
+    /**
+     * @return array<int, string>
+     */
+    public function functions(): array;
+
+    /**
+     * @param array<string, mixed>|Soapable $body
+     */
+    public function call(string $method, array|Soapable $body = []): Response;
```

Comment:
Same as for the Client contract : it is better to make it an array of parameters imo or a variadic